src/pool/srv_pool.c

```diff
@@ -4905,6 +4905,7 @@ ds_pool_upgrade_if_needed(uuid_t pool_uuid, struct rsvc_hint *po_hint,
 	int				rc;
 	bool				schedule_layout_upgrade = false;
 	bool				need_put_leader = false;
```

Comment:
something to consider for code readability (at least would make it easier for me - and hopefully I have it correct below)

need_put_leader if true really means that this function is invoked as a result of the admin running the dmg command to upgrade the pool. And need_put_leader == false means it is running as part of pool service leader step up? So could the variable be named to reflect this? It seems awkward reading the various conditionals in the code and having to always remember that !need_put_leader means the function is running as part of pool step up, and that need_put_leader means the function is running as part of a dmg pool upgrade command invocation.

maybe rename need_put_leader -> dmg_upgrade_cmd ?